src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
```

Comment:
> For some reason I thought that was a strong persistent conversion, which I thought wouldn't leak a handle.

Oh, you're right.  That's an awfully subtle implementation detail though, I'd just create the HandleScope.  You'll need the Context::Scope anyway.